src/python/pants/option/global_options.py

```diff
@@ -1051,6 +1058,43 @@ class BootstrapOptions:
             """
         ),
     )
+    child_process_max_memory_usage = MemorySizeOption(
+        "--child-process-max-memory-usage",
+        advanced=True,
+        default=None,
+        default_help_repr="1GiB",
+        help=softwrap(
+            """
+            The maximum memory usage for all child processes.
+
```

Comment:
This should reference the relation to `child-process-default-memory-usage` as well.